packages/core/src/renderer/getStyleSheetForBucket.ts

```diff
@@ -45,25 +45,31 @@ const styleBucketOrderingMap = styleBucketOrdering.reduce((acc, cur, j) => {
  */
 export function getStyleSheetForBucket(
   bucketName: StyleBucketName,
-  target: Document | undefined,
+  targetDocument: Document | undefined,
+  insertionPoint: HTMLElement | null,
   renderer: GriffelRenderer,
   metadata: Record<string, unknown> = {},
 ): IsomorphicStyleSheet {
   const isMediaBucket = bucketName === 'm';
   const stylesheetKey: StyleBucketName | string = isMediaBucket ? ((bucketName + metadata['m']) as string) : bucketName;
 
   if (!renderer.stylesheets[stylesheetKey]) {
-    const tag: HTMLStyleElement | undefined = target && target.createElement('style');
+    const tag: HTMLStyleElement | undefined = targetDocument && targetDocument.createElement('style');
     const stylesheet = createIsomorphicStyleSheet(tag, bucketName, {
       ...renderer.styleElementAttributes,
       ...(isMediaBucket && { media: metadata['m'] as string }),
     });
 
     renderer.stylesheets[stylesheetKey] = stylesheet;
 
-    if (target && tag) {
-      const elementSibling = findElementSibling(target, bucketName, renderer, metadata);
-      target.head.insertBefore(tag, elementSibling);
+    if (targetDocument && tag) {
+      let beforeElement: Node | null = findElementSibling(targetDocument, bucketName, renderer, metadata);
+
+      if (beforeElement === null && insertionPoint) {
+        beforeElement = insertionPoint.nextSibling;
```

Comment:
I'm not sure this will work as intended? `findElementSibling` returns `null` when the new griffel bucket should be _after_ all other griffel buckets. 

The first griffel bucket inserted will appear after `insertionPoint` as expected. However, if you later add a bucket that should come last, `findElementSibling` will return `null`, and the new one will be inserted directly after `insertionPoint`, and _before_ any other griffel buckets.

Reply:
One option is to have the `insertionPoint` point API instead be an "insert before this node" API. That has a few benefits:
1. The fix to the above problem is simple, change this line to `beforeElement = insertionPoint;` (or a one-liner definition `const beforeElement = findElementSibling(...) || insertionPoint;`)
2. It reduces the chances that style nodes inserted by other libraries will be interleaved with griffel styles. Either the other library inserts at the top of `<head>`, in which case it'll always be before griffel's styles. Or the other library inserts at the bottom of `<head>`, in which case it'll always be after `insertionPoint`.

Reply:
Wow good catch @behowell 👍

I agree that `1.`  will work as expected. It seems that we went with the insertBefore convention already with `findElementSibling` and it makes sense to keep it that way.

Thinking about it a bit more, if we still wante the the insertAfter behaviour, then the `findElementSibling` should return the element _after which to insert the new element_. It's a convention thing 🤷, currently it's internal but once we set the convention with `insertionPoint` we can actually never change it easily 😅.

Also having said that we should rename `findElementSibling` to be more accurate to what kind of sibling it's returning. I thought I originally named the thing pretty clearly, but it's actually quite 💩💩 and we've fallen into our own trap.

Reply:
> One option is to have the insertionPoint point API instead be an "insert before this node" API. That has a few benefits:

@behowell there are valid concerns, however I prefer to keep the same API as at least someone has already (https://github.com/microsoft/griffel/issues/332#issuecomment-1438574739), in this case Emotion's behavior looks nicest to me.

I didn't pay attention for this in the previous implementation 🙈 I reworked it in 27ad3cf and it should work as expected, thanks for heads up 👍

Reply:
Sounds good! Thanks for making the update.